### PR TITLE
Broccoli persistent filter

### PR DESCRIPTION
**Pull request by shokmaster**
- [BREAKING CHANGE] Replaced **broccoli-filter** with [**broccoli-persistent-filter**](https://github.com/stefanpenner/broccoli-persistent-filter).
- Added tests to reach **100% coverage** :v:.
- Replaced deprecated `istanbul` with `nyc`.

### CODE REVIEW

File: lib/htmlbars-minifier.js [hunk 1]
```diff
@@ -1,18 +1,25 @@
-const Filter = require('broccoli-filter');
+const Filter = require('broccoli-persistent-filter');
+const md5Hex = require('broccoli-persistent-filter/lib/md5-hex');
 
-function HtmlbarsMinifier(inputTree, options) {
+function HtmlbarsMinifier(inputNode, options) {
```
Comment (josex2r) [inline lib/htmlbars-minifier.js]: Try to migrate this to ES6 classes and add the property `annotation` to the plugin (debugging purposes). The code seems to be easier to understand using classes:

```js
class Thumbnail extends Filter {
  constructor(inputNode, options = {}) {
    options = Object.assign(options || {}, {
      persist: true,
      annotation: 'HTMLBars Minifier',
    });
    super(inputNode, options);
  }

  // TODO...
}
```

File: lib/htmlbars-minifier.js [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
+	Filter.call(this, inputNode);
 }
 
 HtmlbarsMinifier.prototype = Object.create(Filter.prototype);
 HtmlbarsMinifier.prototype.constructor = HtmlbarsMinifier;
 HtmlbarsMinifier.prototype.extensions = ['hbs'];
 HtmlbarsMinifier.prototype.targetExtension = 'hbs';
+
+HtmlbarsMinifier.prototype.cacheKeyProcessString = function(string, relativePath) {
+	return md5Hex(string.length + 0x00 + relativePath);
```
Comment (josex2r) [inline lib/htmlbars-minifier.js]: Using a cache key with the length of the content in a plain text file can be wrong.
If I have a file with the word `foo` and I change its content to `bar`, the length is the same and the `cacheKey` will not change.

You must use the file content to generate the cache key:

```js
return md5Hex(string + 0x00 + relativePath);
```

File: node-tests/htmlbars-minifier.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+'use strict';
```
Comment (josex2r) [inline node-tests/htmlbars-minifier.js:1]: This tests are fine but you should migrate the `node-tests` to check that the "plugin" is well constructed.

I know this is out of the scope of this task :S Check [this example](https://github.com/BBVAEngineering/broccoli-thumbnail/blob/master/tests/thumbnail-test.js) about using the Broccoli helper.

**Review by josex2r — CHANGES_REQUESTED** [review]

File: lib/htmlbars-minifier.js (hunk 2 above)
Comment (shokmaster) [reply lib/htmlbars-minifier.js]: done

File: node-tests/htmlbars-minifier.js (hunk 3 above)
Comment (shokmaster) [reply node-tests/htmlbars-minifier.js:1]: Yes, is the next goal we've planned to reach :) Thanks!

File: lib/htmlbars-minifier.js (hunk 1 above)
Comment (shokmaster) [reply lib/htmlbars-minifier.js]: Opened the issue #10 with this purpose. Thanks!

**Review by josex2r — DISMISSED** [review]

File: lib/htmlbars-minifier.js (hunk 1 above)
Comment (shokmaster) [reply lib/htmlbars-minifier.js]: Finally I migrated to ES6 classes, and resolved the `persist=true` issue

**Review by josex2r — APPROVED** [review]

**Pull request merged**